src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -391,8 +391,8 @@ public void update(DecoratedKey key, Row oldRow, Row newRow, Memtable memtable,
         MemtableIndex target = liveMemtables.get(memtable);
         if (target == null)
             return;
-
-        ByteBuffer oldValue = getValueOf(key, oldRow, FBUtilities.nowInSeconds());
+        // Use 0 for nowInSecs to get the value from the oldRow to ensure we remove the old value from the index.
```

Comment:
I would add something about TTLs here to make clear this is because we want the old value even if it has already expired.
Also looks like this code is only for vectors?  Could you add a comment in the function about why vectors are different?  Possibly break this code out into an “index vector” method or something? Don’t necessarily have to do that. But this function flow just seems funny to me, and wasn’t immediately clear that this code only applies to vectors.

Reply:
Sure, I'll improve the comment. The code is for vectors now, but https://github.com/datastax/cassandra/pull/1200 makes it for all data types, so I don't think we need to re-organize things in this PR.